cross_compile/Dockerfile_workspace

```diff
@@ -72,34 +74,32 @@ RUN python3 -m pip install -U \
     vcstool
 
 # Install Fast-RTPS dependencies
-RUN apt-get install --no-install-recommends -y \
-    libasio-dev \
-    libtinyxml2-dev
+RUN apt-get update && apt-get install --no-install-recommends -y \
+        libasio-dev \
+        libtinyxml2-dev \
+    && rm -rf /var/lib/apt/lists/*
 
 # Setup ROS2 workspace
+COPY ${ROS2_WORKSPACE}/src /ros2_ws/src
 WORKDIR /ros2_ws
-RUN mkdir -p /opt/ros/${ROS_DISTRO}/
-RUN mkdir -p /opt/ros/${ROS_DISTRO}/share && touch /opt/ros/${ROS_DISTRO}/setup.bash
 
-# Install rosdep dependencies
+# Run rosdep to get dependencies for the copied-in ROS2 workspace
+ENV ROSDEP_SKIP_KEYS="console_bridge fastcdr fastrtps libopensplice67 libopensplice69 rti-connext-dds-5.3.1 urdfdom_headers"
 RUN rm -f /etc/ros/rosdep/sources.list.d/20-default.list    # In case of cached image.
 RUN rosdep init
-RUN rosdep update
-RUN rosdep install --from-paths src /opt/ros/${ROS_DISTRO}/share \
-    --ignore-src \
-    --rosdistro ${ROS_DISTRO} -y \
-    --skip-keys "console_bridge \
-        fastcdr \
-        fastrtps \
-        libopensplice67 \
-        libopensplice69 \
-        rti-connext-dds-5.3.1 \
-        urdfdom_headers"
-
+RUN rosdep update && \
+    apt-get update && \
+    rosdep install --from-paths src \
+        --ignore-src \
+        --rosdistro ${ROS_DISTRO} -y \
+        --skip-keys "$ROSDEP_SKIP_KEYS" \
```

Comment:
Align style and quoting?

Reply:
quotes needed because they are explicitly part of the argument. adding brackets to match style